src/aead/chacha20_poly1305.rs

```diff
@@ -256,29 +244,111 @@ extern {
 
 #[cfg(test)]
 mod tests {
-    use {aead, c, polyfill, test};
-    use super::{ChaCha20_ctr32, CHACHA20_KEY_LEN, make_counter};
-
-    bssl_test!(test_poly1305, bssl_poly1305_test_main);
+    use {aead, test, error, c, polyfill};
+    use super::{ChaCha20_ctr32, CHACHA20_KEY_LEN, POLY1305_STATE_LEN, POLY1305_KEY_LEN,
+                poly1305_init, poly1305_update, poly1305_finish, make_counter};
 
     #[test]
     pub fn test_chacha20_poly1305() {
         aead::tests::test_aead(&aead::CHACHA20_POLY1305,
-            "crypto/cipher/test/chacha20_poly1305_tests.txt");
+                               "crypto/cipher/test/chacha20_poly1305_tests.txt");
     }
 
     #[test]
     pub fn test_chacha20_poly1305_old() {
         aead::tests::test_aead(&aead::CHACHA20_POLY1305_OLD,
-            "crypto/cipher/test/chacha20_poly1305_old_tests.txt");
+                               "crypto/cipher/test/chacha20_poly1305_old_tests.txt");
     }
 
     #[test]
     pub fn test_poly1305_state_len() {
-        assert_eq!((super::POLY1305_STATE_LEN + 255) / 256,
-                    (CRYPTO_POLY1305_STATE_LEN + 255) / 256);
+        assert_eq!((POLY1305_STATE_LEN + 255) / 256,
+                   (CRYPTO_POLY1305_STATE_LEN + 255) / 256);
+    }
+
+    fn test_simd(excess: usize, key: [u8; POLY1305_KEY_LEN],
```

Comment:
Because this is significantly transliterated from Google's code, we need to add a line "// Portions Copyright (c) 2015, Google Inc.", verbatim, underneath the copyright line at the top of this file.